INTMDB-810: [CFN] Updating Troubleshooting with recent issues (#549)
mongodb/mongodbatlas-cloudformation-resources: README.md

Alternatively, you can set the trust relationships of your role via AWS Console: in the IAM dashboard, select your role and click **Trust Relationships**:
![Screenshot 2023-03-31 at 17 32 55](https://user-images.githubusercontent.com/5663078/230436500-fb4ee057-b70e-4580-a94d-f56191728117.png)

> ## Error: 404 (request "INVALID_GROUP_ID") An invalid group ID 6458c1bacde92a3069d69603|default was specified

Reviewer: FYI - this format is going to be extended in the SDKv2 (we provide more details about errors (SDK method etc.)

> ## Error: 404 (request "INVALID_GROUP_ID") An invalid group ID 6458c1bacde92a3069d69603|default was specified

Reviewer 2: ```suggestion
## Error: 404 (request "INVALID_GROUP_ID") An invalid group ID YOUR-PROJECT-ID|default was specified
```

Reviewer 3: I think that it's better to generalize, so YOUR-PROJECT-ID makes sense to me 👍  - you could put it in carrots < > to make it extra clear.

Author: updated

> The problem is caused by using the project resource identifier (ID + Profile Name) as input paramater ProjectID of another CFN resource. The correct approach is to use [GetAttr](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-getatt.html) function to get the ProjectId from the project resource and use that value as input paramenter to the next CFN resource.

Reviewer: ```suggestion
The problem is caused by using the project resource identifier (ID + Profile Name) as the input parameter `ProjectID` of another CFN resource. The correct approach is to use [GetAttr](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-getatt.html) function to get the ProjectId from the project resource and use that value as input parameter to the next CFN resource.
```

Reviewer: Generally headers should be url accesible by hash. It will be good (in future PRs) to make headers specifically about error code etc. Users will be able to search for full the error itself anyway.

Example of correct use of GetAtt (see [project-cluster.json](examples/cluster/project-cluster.json)):
```json
   "AtlasCluster": {
      "Type": "MongoDB::atlas::Cluster",
      "Properties": {
        "ProjectId": {
          "Fn::GetAtt": [
            "Project",
            "Id"
          ]
        },
        "Name": {
          "Ref": "ClusterName"
        },
```

> ## Error: The CFN stack faield to delete `MongoDB::Atlas::NetworkContainer` with the error 409 (request "CONTAINERS_IN_USE") Cannot modify in use containers. The container still contained resources." 

Reviewer 4: ```suggestion
## Error: The CFN stack failed to delete `MongoDB::Atlas::NetworkContainer` with the error 409 (request "CONTAINERS_IN_USE") Cannot modify in use containers. The container still contained resources." 
```

> ## Error: The CFN stack failed to delete `MongoDB::Atlas::NetworkContainer` with the error 409 (request "CONTAINERS_IN_USE") Cannot modify in use containers. The container still contained resources." 

Reviewer 3: [optional/ idea] I think this would be clearer with a period and a new, full sentence 
```suggestion
## Error: The CFN stack failed to delete `MongoDB::Atlas::NetworkContainer` with the error 409 (request "CONTAINERS_IN_USE"). You cannot modify in-use containers; the container still contains resources." 
```

> The problem is caused by having Atlas resources using your network container. If your Atlas project has a cluster or a network peering, you won't be able to delete the network container. 
> To make sure your CFN stack deletes the cluster and the network peering resources before attempting to delete the network container, you should add [DependsOn](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-dependson.html) to your cluster resource.

Reviewer 3: Suggest simplifying with present tense and other tweaks
```suggestion
The problem is that Atlas resources are using your network container. If your Atlas project has a cluster or a network peering resource, you can't delete the network container. 
To make sure your CFN stack deletes clusters and network peering resources before attempting to delete the network container, you should add [DependsOn](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-dependson.html) to your cluster resource.
```
